dll/win32/browseui/shellbrowser.cpp

```diff
@@ -788,9 +788,12 @@ HRESULT CShellBrowser::Initialize()
 
 
     // create status bar
-    fStatusBar = CreateWindow(STATUSCLASSNAMEW, NULL, WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS |
-                    SBT_NOBORDERS | SBT_TOOLTIPS, 0, 0, 500, 20, m_hWnd, (HMENU)0xa001,
-                    _AtlBaseModule.GetModuleInstance(), 0);
+    DWORD dwStatusStyle = WS_CHILD | WS_CLIPSIBLINGS | SBARS_SIZEGRIP | SBARS_TOOLTIPS;
+    if (fStatusBarVisible)
+        dwStatusStyle |= WS_VISIBLE;
+    fStatusBar = ::CreateWindowExW(0, STATUSCLASSNAMEW, NULL, dwStatusStyle,
+                                   0, 0, 500, 20, m_hWnd, (HMENU)FCIDM_TB_UPFOLDER,
+                                   _AtlBaseModule.GetModuleInstance(), 0);
 
     ShowWindow(SW_SHOWNORMAL);
```

Comment:
Is this not going to invalidate the missing WS_VISIBLE?

Reply:
Doesn't this one act on the main window though? Or are you thinking, that by showing the main window it would force all the children (including the status bar) to become visible?

Reply:
Right, I missed that.
Doesn't matter since it is merged anyway.